kratos/geometries/nurbs_curve_on_surface_geometry.h

```diff
@@ -232,6 +233,21 @@ class NurbsCurveOnSurfaceGeometry : public Geometry<typename TSurfaceContainerPo
         return mpNurbsSurface->PolynomialDegree(0) + mpNurbsSurface->PolynomialDegree(1);
     }
 
+    ///@}
+    ///@name Set/ Get functions
+    ///@{
+    /// Returns the NurbsCurve::Pointer of this CurveOnSurface.
+    NurbsCurvePointerType pGetCurve()
```

Comment:
Why do you need that, we have different interfaces for this?

Reply:
We needed to access the data of the underlying curve. If there is a different way to do this, let me know.

Reply:
You can directly use the tessellation of parameter curves.

Reply:
But you still need to expose the DomainInterval and KnotSpanIntervals. Therefore, I chose to expose the underlying curve instead of two methods.

Reply:
That's something that we're currently working on to make it accessable via the geometry interface. The PR is already online.

If everyone is just adding things as you want is becoming a mess. And that's the core we have to be a bit careful.

Reply:
Don't get me wrong. I am not opposed to changing this at all. But I also don't think that adding functionalities that someone needs can be called creating a mess. I can change it so we use the exposed `DomainInterval` from https://github.com/KratosMultiphysics/Kratos/pull/8572, but then there are still the `KnotSpanIntervalls` missing, which are also needed for the tessellation of the curves.